src/atoms/TextInput/TextInput.native.js

```diff
@@ -249,6 +253,12 @@ const TextInput = ({
     [layoutDimensions, setLayoutDimensions],
   );
 
+  useEffect(() => {
+    if (isDefined(value)) {
+      setInput(value);
```

Comment:
`setInput(value||'')`

It should be similar to `useState(value || '')`. Empty string for undefined, just like what RN TextInput does. For undefined values it is empty.

Reply:
That's for input. This only happens when a value is defined or changed. Refer to the declaration of ```input```. That's where we assign default values

Reply:
What I was suggesting is, whatever the new value be just set it(for undefined set it to '') and even if you set `undefined`, RN `TextInput` will be able to handle this. What say?

Reply:
The idea is, that if someone is using it as a controlled TextInput then they _must_ provide value to the ```value``` prop. Providing value as undefined is as good as making it uncontrolled. If you want to clear the text on a controlled TextInput pass the value as ```value=""```

Reply:
If it's made undefined then the value prop holds no value and the component should be assumed to be uncontrolled